packages/coinstac-ui/app/render/components/computation-field-covariates.js

```diff
@@ -0,0 +1,146 @@
+import React, { Component, PropTypes } from 'react';
+import {
+  Button,
+  ControlLabel,
+  FormControl,
+  FormGroup,
+  HelpBlock,
+} from 'react-bootstrap';
+
+export default class ComputationFieldCovariates extends Component {
+  constructor(props) {
+    super(props);
+
+    this.handleNewClick = this.handleNewClick.bind(this);
+    this.getHandleValueChange = this.getHandleValueChange.bind(this);
+  }
+
+  getHandleValueChange(property, index) {
```

Comment:
Higher order function to retain an `index` and use the event. Could be replaced with something like:

```js
things.map((thing, index) => {
  return (
    <input
      onChange={(event) => {
        this.props.onChange({ 'my-manual-propery': event.target.value }, index);
      }}
    />
  );
});
```